### PR TITLE
Implemented a mtg search function as well, together with fixing concerns

### PR DESCRIPTION
**Pull request by bumbadadabum**
A lot of duplicate code though, but I don't see a way to do it better.

### CODE REVIEW

File: chat-plugins/tcg.js [hunk 1]
```diff
@@ -1,21 +1,52 @@
...
 	ygo: 'yugioh',
 	yugioh: function (target, room, user) {
 		if (room.id !== 'tcg') return this.sendReply("This command can only be used in the TCG room.");
 		if (!this.canBroadcast()) return;
 
-		var http = require("http");
+		var query = Tools.escapeHTML(target);
 		var self = this;
 
+		var Search = http.get(('http://yugioh.wikia.com/api/v1/Search/List/?query=' + query + '&limit=1'), function (a) {
```
Comment (kotarou3) [inline chat-plugins/tcg.js]: Unnessasary brackets around the URI. Same for below occurance

**Comment by ascriptmaster** [comment]
To remove the duplicate code, alias all four commands to a single function with extra arguments like this: `function (target, room, user, connection, cmd)`

`cmd` will then let you check the exact command used (typing `/ygo Kuriboh` has `cmd` equal `'ygo'`), and direct the HTTP Request to the appropriate Wikia page.

**Comment by Zarel** [comment]
You don't actually fix a lot of the problems. :|

**Comment by bumbadadabum** [comment]
I'm probably going to make it all neat and nice whenever I can, just don't have the time nor knowledge to this this correctly.

**Comment by Zarel** [comment]
If you don't have the time nor knowledge to do it correctly, the correct answer seems to be to delete your plugin.

**Comment by bumbadadabum** [comment]
Says the person that merged it.

**Comment by bumbadadabum** [comment]
Yeah sorry I shouldn't attempt sarcasm. Well anyway I see it can be done better, and would have been fine if you didn't merge it until it was better. I would probably have time this weekend to do that, if that's still wanted.

**Comment by Zarel** [comment]
It's not good, but it's better than what was there before, which is why I merged it.

**Comment by bumbadadabum** [comment]
Yeah I'll make another PR this weekend then. Do notify me of all the concerns you have with it and I'll try my best to fix everything.

**Pull request merged**